synapse/lib/ast.py

```diff
@@ -2950,14 +2964,32 @@ async def run(self, runt, genr):
         async for node, path in genr:
 
             name = await self.kids[0].compute(runt, path)
-
-            valu = await self.kids[2].compute(runt, path)
-            valu = await s_stormtypes.toprim(valu)
+            #valu = await self.kids[2].compute(runt, path)
 
             prop = node.form.props.get(name)
             if prop is None:
                 raise s_exc.NoSuchProp(name=name, form=node.form.name)
 
+            expand = True
+            isarray = isinstance(prop.type, s_types.Array)
+
+            valu = await self.kids[2].compute(runt, path)
+            valu = await s_stormtypes.toprim(valu)
+
+            # Setting a value to a subquery means assigning the value of
+            # the primary property (for a single value) or array of values
+            if isinstance(self.kids[2], SubQuery):
+
+                if not isarray:
+                    if len(valu) != 1:
+                        mesg = 'Cannot assign a property value from multiple nodes.'
```

Comment:
This error message could be confusing if they get it due to the subquery returning no nodes.